daemon/names.go

```diff
@@ -69,7 +69,7 @@ func (daemon *Daemon) reserveName(id, name string) (string, error) {
 				logrus.Errorf("got unexpected error while looking up reserved name: %v", err)
 				return "", err
 			}
-			return "", validationError{errors.Errorf("Conflict. The container name %q is already in use by container %q. You have to remove (or rename) that container to be able to reuse that name.", name, id)}
+			return "", nameConflictError{errors.Errorf("Conflict. The container name %q is already in use by container %q. You have to remove (or rename) that container to be able to reuse that name.", name, id)}
```

Comment:
Almost, but we don't need the wrapped `errors.Errorf` here.
We can generate the message in the `Error()` function and store the conflicting name and ID in the error struct.

Reply:
Ah! Got it; wasn't sure to what extend you wanted to differentiate from a plain conflictError